File: App/Backend/src/models/course.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from Backend.database import Base


class Course(Base):
    __tablename__ = "courses"

    course_id: Mapped[int] = mapped_column(
        Integer,
        primary_key=True,
        autoincrement=True
    )

    course_name: Mapped[str] = mapped_column(
        String(100),
        nullable=False
    )

    description: Mapped[str | None] = mapped_column(
        Text,
        nullable=True
    )

    duration: Mapped[str | None] = mapped_column(
        String(50),
        nullable=True
    )

    status: Mapped[str] = mapped_column(
        String(20),
        default="draft"
    )

    category: Mapped[str | None] = mapped_column(
        String(100),
        nullable=True
    )

    created_by: Mapped[str | None] = mapped_column(
        String(10),
        ForeignKey("users.uid"),
        nullable=True
    )

    published_by: Mapped[str | None] = mapped_column(
        String(10),
        ForeignKey("users.uid"),
        nullable=True
    )

    created_at: Mapped[datetime | None] = mapped_column(
        DateTime,
        default=datetime.utcnow
    )

    updated_at: Mapped[datetime | None] = mapped_column(
        DateTime,
        nullable=True
    )


class CourseTeacher(Base):
    __tablename__ = "course_teachers"

    course_id: Mapped[int] = mapped_column(
        Integer,
        ForeignKey("courses.course_id"),
        primary_key=True
    )

    teacher_id: Mapped[int] = mapped_column(
        Integer,
        ForeignKey("teachers.teacher_id"),
        primary_key=True
    )

    is_course_admin: Mapped[bool] = mapped_column(
        Boolean,
        default=False,
        nullable=False
    )

File: App(SpringBoot)/src/main/java/com/Project/LearningManagementSystem/entity/Course.java
package com.Project.LearningManagementSystem.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "courses")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Course {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "course_id")
    private Integer courseId;

    @Column(name = "course_name", length = 100, nullable = false)
    private String courseName;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "duration", length = 50)
    private String duration;

    @Column(name = "status", length = 20, nullable = false)
    private String status = "draft";

    @Column(name = "category", length = 100)
    private String category;

    @Column(name = "created_by", length = 10)
    private String createdBy;

    @Column(name = "published_by", length = 10)
    private String publishedBy;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
